### ISSUE TITLE
Not maintained ?

### ISSUE THREAD

**Issue by cedvan** [tantaman/Strut#380]
Hi, 

The work done is great! But these repositories are no longer maintained? If yes, why !?

**Comment by ejvindh** [comment]
Strut was a one-person project -- probably the developer (tantaman) got busy doing other stuff, or simply lost interest :)

**Comment by tantaman** [comment]
After a 4 year break, I'm coming back to the project.

**Comment by MohannadNaj** [comment]
Hi @tantaman 

It's great seeing you back to the project!

I was learning while trying to enhance many aspects of Strut, my code is a mess, I admit it, but the end user-facing results may looks good. It's my first-and-only backbone/requirejs/,,, project, but I tried as much as I can to follow your structure.

You can check it out here:
https://github.com/MohannadNaj/Strut

and the dist version at `gh-pages`:
https://mohannadnaj.github.io/Strut/

Mainly the newly added features:

- Migrated to Bootstrap 3 (only this part is included in this PR https://github.com/tantaman/Strut/pull/375 although I remember it's not completed and I had to modify some files here and there after the latest commit on that PR).

- Ability to paste images directly from the keyboard in the "Image" modal!

- Ability to move components using the keyboard (with/without Shift for larger movement, right /left /up /down)

- Adding substeps support!

With a z-index to increase/decrease, a feature for better arrangement of components inside the slide and showing one substep before the other. This controls in the upper left corner:

![Screenshot of substeps controls](https://i.imgur.com/FHwsg5n.png)

- Adding and using "Settings" control button
![Screenshot of settings](https://i.imgur.com/cqjjUOL.png)

As a control modal for some user settings

using this modal, the options for now:
[ ] Upload Images to Imgur.com `(ability to use base64 instead)`
[ ] Show usage hint at presentation beginning `(the first hint in the generated presentation)`
[ ] Hide other slides while navigating `(before this, excluding the active slide all slides in impress had a opacity high degree, this setting will hide it completely).
[ ] Hide substeps only before the first time they enter the slide, then keep it shown `(prevents hiding the newly showed substeps after stepping out of a slide)

----

I'm not confident enough to submit all of these changes as a PR. it's alot of changes.

If there is some specific features up there you want me to extract and submit alone, I'm happy to do it. otherwise, there is a possibility that my ideas may inspire you. and I'm really excited seeing you back here, I do believe in your project/philosophy!

**Comment by tantaman** [comment]
MohannadNaj - the substep feature is really cool. I'll see if I can merge your changes back into the main repo.

**Comment by tantaman** [comment]
After refreshing my memory on both codebases (strut and strut2) further development will be happening on Strut2.

You can see the vision for Strut2 as it comes together here: https://github.com/tantaman/strut2/blob/master/README.md

**Comment by tantaman** [comment]
Strut2 is being deployed here: https://tantaman.com/strut2/ (only supports authoring slides at the moment)

until it's ready for release. Note that the editor is markdown based.

**Comment by MohannadNaj** [comment]
@tantaman That's Awesome!

Wish I can do more to this project -Just to let you know, I used it to generate my graduation project presentation back at the end of 2017, which means I'll never forget Strut-.. the time will come when I'll be able to do more -I mean Frontend contributions-.

I guess now I'll just take the backseat and cheer: Go Matthew, Go!

Also, I think you already have a better plans for the backend subscription service. If not and if you are ok with PHP -Laravel framework- I'll be more than happy to work on that to take away some of the Backend work.

![image](https://user-images.githubusercontent.com/8743700/109430820-45910a80-7a14-11eb-8d23-6efa26d46e86.png)

**Comment by tantaman** [comment]
Just a heads up that a new version has been deployed which allows:
- saving
- undo/redo
- linking slides to one another via `@` mentioning content from a slide
- different styles
- canned layouts and transitions for presenting

**Comment by TempleOfTemplar** [comment]
could you please share the current WIP branch of strut2, if it possible?

**Comment by TempleOfTemplar** [comment]
what's the current progress about Strut 2?

**Comment by tantaman** [comment]
Was working on offline support as well as multiplayer & collaborative editing and fell down a rabbithole.

tldr is that there isn't great existing tech to support those features. The opportunity was too good to pass up and I've been building https://vlcn.io/ for the last ~4-5 months to solve that problem.

Strut will resume as a use case of vlcn after the vlcn.io launch.
